Apps/Examples/Examples/All Examples/PitchAndDistanceExample.swift

```diff
@@ -0,0 +1,85 @@
+// field of a symbol layer to remove large size POI labels in the far
+// distance at high pitch, freeing up that screen real-estate for smaller road and street labels.
+
+import Foundation
+import MapboxMaps
+
+@objc(PitchAndDistanceExample)
+final class PitchAndDistanceExample: UIViewController, ExampleProtocol {
+    private var mapView: MapView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let mapInitOptions = MapInitOptions(
+            cameraOptions: CameraOptions(
+                center: CLLocationCoordinate2D(
+                    latitude: 38.888,
+                    longitude: -77.01866),
+                zoom: 15,
+                pitch: 75),
+            styleURI: StyleURI.streets)
+        mapView = MapView(frame: view.bounds, mapInitOptions: mapInitOptions)
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        mapView.ornaments.options.scaleBar.visibility = .visible
+
+        view.addSubview(mapView)
+        // Wait for the map to load its style before setting the filter.
+        mapView.mapboxMap.onNext(event: .mapLoaded) { _ in
+            self.setPitchDistanceFilter()
+
+            // The below line is used for internal testing purposes only.
+            self.finish()
+        }
+    }
+
+    func updateFilter(currentFilter: Expression) -> Expression {
+        // Add an additional condition to the current filter
+        // to filter based on ["pitch"] and ["distance-from-center"]
+        let updatedFilter = Exp(.all) {
+            currentFilter
+            Exp(.switchCase) {
+                // Always show the symbol when pitch <= 60
+                Exp(.lte) {
+                    Exp(.pitch)
+                    60
+                }
+                true
+                // When pitch > 60, show the symbol only
+                // when it is close to the camera ( distance <= 2 )
+                Exp(.all) {
+                    Exp(.lte) {
+                        Exp(.distanceFromCenter)
+                        2
+                    }
+                    Exp(.gt) {
+                        Exp(.pitch)
+                        60
+                    }
+                }
+                true
+                // Hide in the remaining case, far and high pitch
+                false
+            }
+        }
+        return updatedFilter
+    }
+
+    func setPitchDistanceFilter() {
+        let poiLayers = ["poi-label", "transit-label"]
+
+        for layerID in poiLayers {
+            do {
+                try mapView.mapboxMap.style.updateLayer(withId: layerID, type: SymbolLayer.self, update: { (layer: inout SymbolLayer) in
+                    guard let currentFilter = layer.filter else {
+                        return
+                    }
+                    layer.filter = updateFilter(currentFilter: currentFilter)
+                })
+            } catch {
+                print("Updating the layer failed: \(error.localizedDescription)")
```

Comment:
Nit: It might have sense to use access to the layerID variable in this context:
```suggestion
                print("Updating the layer '\(layerID)' failed: \(error.localizedDescription)")
```